### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by hemangsk**

### CODE REVIEW

File: projects/ngx-morse/src/lib/ngx-morse.service.spec.ts [hunk 1]
```diff
@@ -1,12 +1,83 @@
...
+    const message: string = 'Keep the dream alive\t\t\t';
+    expect(() => {
+      // https://stackoverflow.com/a/4144803/5056792
+      ngxMorseService.encode(message);
+    }).toThrow(new InvalidInputError(InvalidInputError.INVALID_INPUT));
+  });
+
+  // Decode tests
+
+  it('decode: should skip decoding an empty string', () => {
```
Comment (hemangsk) [inline projects/ngx-morse/src/lib/ngx-morse.service.spec.ts:59]: Currently ngxMorseService.decode(< empty string >) returns undefined. We can also throw an invalid input error in that case. what are your thoughts on this @monkeyscript ?

File: projects/ngx-morse/src/lib/ngx-morse.service.spec.ts [hunk 2]
```diff
@@ -1,12 +1,83 @@
...
+    const emptyString: string = '';
+    expect(() => {
+      ngxMorseService.decode(emptyString)
+    }).toThrow(new InvalidInputError(InvalidInputError.INVALID_MORSE_INPUT));
+  });
+
+  it('decode: should decode morse code generated from valid input', () => {
+    const encodedMessage: string = '-.- . . .--. / - .... . / -.. .-. . .- -- / .- .-.. .. ...- . ';
+    const expectedMessageResult: string = 'Keep the dream alive'.toLowerCase();
+    // TODO: Discuss about keeping/removing whitespace at end of results
```
Comment (hemangsk) [inline projects/ngx-morse/src/lib/ngx-morse.service.spec.ts:72]: The output of both functions seem to have trailing whitespace. Is that intended? Or should it be trimmed. I'd suggest we trim the output in the encode / decode method before returning it.

File: projects/ngx-morse/src/lib/ngx-morse.service.spec.ts (hunk 1 above)
Comment (monkeyscript) [reply projects/ngx-morse/src/lib/ngx-morse.service.spec.ts:59]: Yeah. It's a valid scenario.

File: projects/ngx-morse/src/lib/ngx-morse.service.spec.ts (hunk 2 above)
Comment (monkeyscript) [reply projects/ngx-morse/src/lib/ngx-morse.service.spec.ts:72]: I've seen many examples with trailing spaces. But now I'm having second thoughts. Needs some more research.

**Review by monkeyscript — APPROVED** [review]

**Comment by monkeyscript** [comment]
@hemangsk I'm merging this for now. If you have any changes or improvements in your mind, feel free to open a new PR.

**Pull request merged**